htdocs/mri/jiv/get_file.php

```diff
@@ -140,40 +122,93 @@
     $MimeType = 'application/octet-stream';
     break;
 case 'xml':
-    $FullPath         = $imagePath . '/' . $File;
-    $MimeType         = 'application/xml';
-    $DownloadFilename = basename($File);
+    $FullPath = $imagePath . '/' . $File;
+    $MimeType = 'application/xml';
     break;
 case 'nrrd':
-    $FullPath         = $imagePath . '/' . $File;
-    $MimeType         = 'image/vnd.nrrd';
-    $DownloadFilename = basename($File);
+    $FullPath = $imagePath . '/' . $File;
+    $MimeType = 'image/vnd.nrrd';
     break;
 case 'DICOMTAR':
     // ADD case for DICOMTAR
-    $FullPath         = $tarchivePath . '/' . $File;
-    $MimeType         = 'application/x-tar';
-    $DownloadFilename = basename($File);
+    $FullPath = $tarchivePath . '/' . $File;
+    $MimeType = 'application/x-tar';
     break;
 default:
-    $FullPath         = $DownloadPath . '/' . $File;
-    $MimeType         = 'application/octet-stream';
-    $DownloadFilename = basename($File);
+    $FullPath = $DownloadPath . '/' . $File;
+    $MimeType = 'application/octet-stream';
     break;
 }
 
+// Make sure file exists.
 if (!file_exists($FullPath)) {
-    error_log("ERROR: File $File does not exist");
-    header("HTTP/1.1 404 Not Found");
-    exit(5);
+    error_log("ERROR: Requested file $FullPath does not exist");
+    http_response_code(404);
+    return;
 }
 
+// Build and send the response with the file data.
 header("Content-type: $MimeType");
-if (!empty($DownloadFilename)) {
-
-    header("Content-Disposition: attachment; filename=$DownloadFilename");
-}
+header('Content-Disposition: attachment; filename=' . basename($File));
 $fp = fopen($FullPath, 'r');
 fpassthru($fp);
 fclose($fp);
-?>
+
+/**
+ * Checks that config settings used are not empty and not the root dir.
+ *
+ * @param array $paths The values to validate.
+ *
+ * @return bool
+ */
+function validConfigPaths(array $paths): bool
+{
+    foreach ($paths as $p) {
+        if (empty($p)) {
```

Comment:
@johnsaigle This is not working for me. I've got imagePath, mincPath, tarchiveLibraryDir and downloadPath properly set and I keep running into the following error:
```
[Mon Nov 19 10:39:35.394047 2018] [php7:notice] [pid 13763] [client 132.216.237.71:53063] PHP Notice:  Undefined index: imagePath in /var/www/loris/htdocs/mri/jiv/get_file.php on line 46, referer: https://cecile-dev.loris.ca/imaging_browser/viewSession/?sessionID=2566&outputType=native&backURL=/imaging_browser/
[Mon Nov 19 10:39:35.394432 2018] [php7:notice] [pid 13763] [client 132.216.237.71:53063] PHP Notice:  Undefined index: DownloadPath in /var/www/loris/htdocs/mri/jiv/get_file.php on line 47, referer: https://cecile-dev.loris.ca/imaging_browser/viewSession/?sessionID=2566&outputType=native&backURL=/imaging_browser/
[Mon Nov 19 10:39:35.394709 2018] [php7:notice] [pid 13763] [client 132.216.237.71:53063] PHP Notice:  Undefined index: mincPath in /var/www/loris/htdocs/mri/jiv/get_file.php on line 48, referer: https://cecile-dev.loris.ca/imaging_browser/viewSession/?sessionID=2566&outputType=native&backURL=/imaging_browser/
[Mon Nov 19 10:39:35.395027 2018] [php7:error] [pid 13763] [client 132.216.237.71:53063] PHP Fatal error:  Uncaught LorisException: Config paths are not initialized. Please ensure that valid paths are set for imagePath, downloadPath, mincPath and tarchiveLibraryDir in /var/www/loris/htdocs/mri/jiv/get_file.php:168\nStack trace:\n#0 /var/www/loris/htdocs/mri/jiv/get_file.php(56): validConfigPaths(Array)\n#1 {main}\n  thrown in /var/www/loris/htdocs/mri/jiv/get_file.php on line 168, referer: https://cecile-dev.loris.ca/imaging_browser/viewSession/?sessionID=2566&outputType=native&backURL=/imaging_browser/
```

Reply:
```
+--------------------+------------------------------------+
| Name               | Value                              |
+--------------------+------------------------------------+
| DownloadPath       | /data/preventAD/data/userdownloads |
| imagePath          | /data/preventAD/data/              |
| mincPath           | /data/preventAD/data/              |
| tarchiveLibraryDir | /data/preventAD/data/tarchive/     |
+--------------------+------------------------------------+
```

Reply:
Did some more digging. It looks like it is working when I set mincPath, imagePath and DownloadPath in config.xml. So it works if the paths are set in config.xml for the "paths" section.

If I remove all paths from project/config.xml and have the paths correctly set in the Config table, I get the error above. 

Hopefully that will help finding out where the issue comes from.

Reply:
@johnsaigle Nevermind. There was an issue with my sandbox. Forget my comments. Sorry...